pipeline/stage05_wave_characterization/scripts/local_macro_observables.py

```diff
@@ -0,0 +1,103 @@
+import neo
+import numpy as np
+import quantities as pq
+import matplotlib.pyplot as plt
+import os
+import argparse
+import scipy
+import pandas as pd
+from utils import load_neo, none_or_str, save_plot, AnalogSignal2ImageSequence
+import math 
+
+
+def calc_point_velocities(evts, DIM_X, DIM_Y):
+    N_Waves = len(np.unique(evts.labels)) 
+    WaveColl = np.ones([N_Waves, DIM_X, DIM_Y])*np.nan
+    WaveColl[np.int32(np.float64(evts.labels)), np.int32(evts.array_annotations['x_coords']), np.int32(evts.array_annotations['y_coords'])] = evts.times
```

Comment:
Create a three dimensional array, with 100x100 pixels X nbr of waves, quickly explodes for long recordings, and will take up 100GB + of memory. It's better to loop over the waves here (and only slightly slower)